### PR TITLE
feat: add rewards multiplier to the vesting engine

### LINKED ISSUE #9147: implement rewards bonus multiplier
https://github.com/vegaprotocol/specs/pull/1876/files#diff-4f65d45e80a36063c0d6fb245ba582f3bd5e5d7f060c414cd723e97df67bf794R33

### PR DESCRIPTION
**Pull request by jeremyletang**
close #9147

### CODE REVIEW

File: protos/sources/vega/vega.proto [hunk 1]
```diff
@@ -1621,3 +1621,12 @@ message BenefitTier {
...
   string referral_discount_factor = 4;
 }
+
+message VestingBenefitTiers {
+  repeated VestingBenefitTier tiers = 1;
+}
+
+message VestingBenefitTier {
+  string minimum_quantum_balance = 1;
+  string reward_multiplier = 2;
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:1631]: Please add doc strings here

Comment (jeremyletang) [reply protos/sources/vega/vega.proto:1631]: this is never used in an API really, it's uncessary...

Comment (barnabee) [reply protos/sources/vega/vega.proto:1631]: Is it not used when the community want to update the tiers?

**Review by barnabee — DISMISSED** [review]

File: protos/sources/vega/vega.proto (hunk 1 above)
Comment (jeremyletang) [reply protos/sources/vega/vega.proto:1631]: not really, only internally as network parameters are all just strings

File: core/collateral/engine.go [hunk 2]
```diff
@@ -131,6 +131,26 @@ func New(log *logging.Logger, conf Config, ts TimeService, broker Broker) *Engin
...
 }
 
+func (e *Engine) GetAllVestingQuantumBalance(party string) *num.Uint {
+	balance := num.UintZero()
+
+	for asset, details := range e.enabledAssets {
+		// vesting balance
+		if acc, ok := e.accs[e.accountID(noMarket, party, asset, types.AccountTypeVestingRewards)]; ok {
+			quantumBalance, _ := num.UintFromDecimal(acc.Balance.ToDecimal().Div(details.Details.Quantum))
+			balance.Add(balance, quantumBalance)
```
Comment (ze97286) [inline core/collateral/engine.go:141]: balance.AddSum(quantumBalance)

**Review by guoguojin — DISMISSED** [review]

**Review by EVODelavega — DISMISSED** [review]

File: core/collateral/engine.go [hunk 3]
```diff
@@ -131,6 +131,26 @@ func New(log *logging.Logger, conf Config, ts TimeService, broker Broker) *Engin
...
+		}
+
+		// vested balance
+		if acc, ok := e.accs[e.accountID(noMarket, party, asset, types.AccountTypeVestedRewards)]; ok {
+			quantumBalance, _ := num.UintFromDecimal(acc.Balance.ToDecimal().Div(details.Details.Quantum))
+			balance.Add(balance, quantumBalance)
```
[the remark this answers is not in the document]
Comment (jeremyletang) [reply core/collateral/engine.go]: addressed

File: core/collateral/engine.go (hunk 2 above)
Comment (jeremyletang) [reply core/collateral/engine.go]: addressed

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**